app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,97 @@
 class SendgridApi
-  extend SingleForwardable
+  TIMEOUT = 2 # seconds
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  RESCUABLE_ERRORS = [SendgridClient::Error,
+                      JSON::ParserError,
+                      Excon::Errors::Error,
+                      Timeout::Error
+                     ].freeze
 
   def spam_reported?(email)
-    api { spam_reports.retrieve(email: email).any? }
+    action = 'spamreports.get.json'
+    query = { email: email }
+
+    response = SendgridPool.instance.with { |conn|
+      conn.get_request(action, query)
+    }
+
+    return false if error_response?(response)
+
+    response.any?
```

Comment:
I removed the rescue of `StandardError` as suggested previously, however I'd like to put it back in to make the Sendgrid integration inside the request more robust.

I would do it a bit differently than before though, as part of rescuing the error I'd send the error to Sentry so that we get notified of unexpected errors. What are thoughts on doing this?